xmtp/src/association.rs

```diff
@@ -83,7 +83,7 @@ impl Association {
         let addr = wallet.get_address();
         let assoc_text = AssociationText::new_static(addr, pub_key);
 
-        let signature = wallet.sign(assoc_text.clone())?;
+        let signature = wallet.sign(&assoc_text.text())?;
```

Comment:
This makes sense to me. Might as well reduce the complexity of any types going over the FFI boundary